src/commands/misc/dog.js

```diff
@@ -11,9 +11,12 @@ module.exports = class Dog extends Command {
   run (message) {
     message.channel.startTyping()
     request.get('https://random.dog/woof.json').then(r => {
-      const embed = this.client.getDefaultEmbed(message.author)
-      embed.setImage(r.body.url).setDescription('Here is your dog <:DoggoF:445701839564963840>')
-      message.channel.send({embed})
+      var notsupported = ['.mp4']
+      if (!r.body.url.endsWith(notsupported)) {
+        const embed = this.client.getDefaultEmbed(message.author)
+        embed.setImage(r.body.url).setDescription('Here is your dog <:DoggoF:445701839564963840>')
+        message.channel.send({embed})
+      } else this.run(message)
     })
     message.channel.stopTyping()

```

Comment:
This should be inside of the request.then method.

Reply:
You should probably use async function too, it makes the code cleaner.